src/onegov/page/model.py

```diff
@@ -40,12 +41,12 @@ def root(self) -> Page: ...
         def ancestors(self) -> Iterator[Page]: ...
 
         # HACK: Workaround for hybrid_property not working until SQLAlchemy 2.0
-        published_or_created: Column[bool]
+        published_or_created: Column[datetime]
     else:
         @hybrid_property
-        def published_or_created(self):
+        def published_or_created(self) -> datetime:
             return self.publication_start or self.created
 
         @published_or_created.expression  # type:ignore[no-redef]
-        def published_or_created(cls):
+        def published_or_created(cls) -> datetime:
```

Comment:
```suggestion
        def published_or_created(cls) -> ColumnElement[datetime]:
```

Would be the correct annotation here, but it's in a block that's not evaluated by the type checker, so there isn't much of a point to adding the annotations here.